main.py

```diff
@@ -0,0 +1,51 @@
+import time
+import os
+import psutil
+
+
+def elapsed_since(start):
+    return time.strftime("%H:%M:%S", time.gmtime(time.time() - start))
+
+
+def get_process_memory():
+    process = psutil.Process(os.getpid())
+    mem_info = process.memory_info()
+    return mem_info.rss
+
+
+def profile(func):
+    def wrapper(*args, **kwargs):
+        mem_before = get_process_memory()
+        start = time.time()
```

Comment:
process = psutil.Process() < it will use by default the current process.
mem_before = process.memory_info().rss
result = func(*args, **kwargs)
mem_after = process.memory_info().rss

the same rows count without function